zookeeper-server/src/main/java/org/apache/zookeeper/server/watch/WatchManager.java

```diff
@@ -305,13 +322,12 @@ public synchronized WatchesSummary getWatchesSummary() {
     @Override
     public void shutdown() { /* do nothing */ }
 
-    @Override
-    public int getRecursiveWatchQty() {
-        return watcherModeManager.getRecursiveQty();
+    int getRecursiveWatchQty() {
```

Comment:
Missing synchronised here

Reply:
Done. But I think it is not necessary as it is visible for testing. It is only accessed in `getPathParentIterator` which has been moved under `synchronized`. Anyway, it is not harm.